rig-lancedb/src/utils/mod.rs

```diff
@@ -26,3 +26,58 @@ impl QueryToJson for lancedb::query::VectorQuery {
         record_batches.deserialize()
     }
 }
+
+pub trait FilterEmbeddings {
+    fn filter(&mut self, embeddings_col: Option<String>) -> serde_json::Result<serde_json::Value>;
+}
+
+impl FilterEmbeddings for serde_json::Value {
+    fn filter(&mut self, embeddings_col: Option<String>) -> serde_json::Result<serde_json::Value> {
```

Comment:
Either `self` is mutable and `filter` modifies the object inplace, or `self` is not mutable and a copy is returned. Right now it's both which is not good practice